### PR TITLE
move authorized keys and user config into environments

### PR DESCRIPTION
**Pull request by dannyroberts**
Our current way of specifying what user accounts and authorized_keys to add to the VMs for an environment is too hard-coded for dimagi. The goal in this refactor is to preserve the shared nature of user config (i.e. `swiss`, `icds`, `production`, etc. all may want to have the same user accounts and keys) and keep things DRY, while still giving a different team the ability to swap in their own users and keys in a first-class way / the same way that we do. This PR takes this strategy:

- inside `environments/_users` you can store your user config that currently contains just the `dev_users` field that we used to store in `all.yml`. Name the file `<group>.yml` where "group" is a nickname/id for the group of users in that file. Ours for example is called `dimagi.yml`
- inside `environments/_authorized_keys` you can store all of the .pub files. All user configs go off the same shared set authorized keys, so it's easy to have overlap between the user config files without having to duplicate the keys.
- inside each env's `meta.yml`, add `users: <group>` to tell the env which users file to go off of

This gets us one big step closer to being team-agnostic, i.e. letting others swap in their environment configurations in a way that's totally parallel to our example dimagi one that's committed in this repo.

### CODE REVIEW

File: commcare-cloud/commcare_cloud/environment/paths.py [hunk 1]
```diff
@@ -57,6 +57,14 @@ def fab_settings_yml_default(self):
     def generated_yml(self):
         return os.path.join(ENVIRONMENTS_DIR, self.env_name, '.generated.yml')
 
+    @lazy_immutable_property
+    def authorized_keys_dir(self):
+        return os.path.join(ENVIRONMENTS_DIR, '_authorized_keys')
+
+    @memoized
```
[the remark this answers is not in the document]
Comment (dannyroberts) [reply commcare-cloud/commcare_cloud/environment/paths.py]: Oh, because this is an actual method (with a non-self arg) and not a property.

**Comment by snopoke** [comment]
👍

**Comment by czue** [comment]
@dannyroberts I think I'm just revealing my ansible ignorance, but do you have any guidance on how to get this working with https://github.com/dimagi/ansible-minimal? I'm getting an error: `fatal: [emm.dimagi.com]: FAILED! => {"msg": "'dev_users' is undefined"}`.

If I copy `dev_users` in to my site's public vars it works, but I haven't figured out the right way to get the `meta.yml` to reference the `dimagi` group successfully.

(feel free to also tell me that supporting non cchq deployments is out of scope for what you're trying to do )

**Comment by dannyroberts** [comment]
Ha! Right, I definitely wasn't thinking about this, but I wonder if it's possible to get working. Do you currently have a directory set up like this?

```
<environments>/
    <env>/
        inventory.ini
        public.yml
        ...
```
? If that's been working in the past, then try adding `meta.yml` there (which it sounds like you tried), and then also `_authorized_keys` and `_users` directories under `<environments>` (in the repo it's called "environments", but I put it in angle brackets because you can call it anything). It'd end up something like this

```
<environments>/
    _authorized_keys/
        czue.pub
        wpride.pub
        ...
    _users/
        <user_group>.yml  # dev_users: ...
    <env>/
        inventory.ini
        public.yml
        meta.yml  # users: <user_group>
```

**Pull request merged**